core/src/main/scala/org/apache/spark/scheduler/TaskSetManager.scala

```diff
@@ -1218,6 +1262,53 @@ private[spark] class TaskSetManager(
   def executorAdded(): Unit = {
     recomputeLocality()
   }
+
+  /**
+   * A class for checking inefficient tasks to be speculated, a task is inefficient when its data
+   * process rate is less than the average data process rate of all successful tasks in the stage
+   * multiplied by a multiplier.
+   */
+  private[TaskSetManager] class TaskProcessRateCalculator {
+    private var totalRecordsRead = 0L
+    private var totalExecutorRunTime = 0L
+    private var avgTaskProcessRate = 0.0D
```

Comment:
Shall we initialize `avgTaskProcessRate` with `Double.MaxValue` so that `taskProcessRateIsInefficient` can be simplified as https://github.com/apache/spark/pull/36162/files#r903856288?

Reply:
done.